### PR TITLE
Feat/tvm example

### PR DESCRIPTION
**Pull request by vilukissa68**
Adds example for running TVM tinyperf benchmark on Headsail. #69 needs to be merged before this.

### CODE REVIEW

**Comment by vilukissa68** [comment]
#72 and #73 need to be merged before this

File: examples/hpc-c/tvm-hpc/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+
+## Python dependencies
+Python dependencies are needed for building TVM models from onnx graphs and must be available during tvm-hpc compilation. 
+
+Install python dependencies for TVM
+``` sh
+cd tvm/python
+python gen_requirements.py
+pip install -r requirements/core.txt
```
Comment (hegza) [inline examples/hpc-c/tvm-hpc/README.md:58]: Nice. I was getting ready to comment to add a requirements.txt but you've already put it in.

File: examples/hpc-c/tvm-hpc/requirements.txt
Comment (hegza) [inline examples/hpc-c/tvm-hpc/requirements.txt:1]: Consider annotating the exact versions used as well. These are known to get stale.

File: examples/hpc-c/tvm-hpc/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+# Add the executable target
+add_executable(${CMAKE_PROJECT_NAME} ${SOURCE_FILES} ${LIB_PATH}/crt0.o)
+
+# Link newlib
+target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE ${LIB_PATH}/libc.a)
+target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE ${LIB_PATH}/libm.a)
+target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE ${LIB_PATH}/libgloss.a)
+target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE -lc -lm -lgloss -lgcc)
+
+target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE /Users/vainogranat/work/headsail-newlib/libgloss/riscv)
```
Comment (hegza) [inline examples/hpc-c/tvm-hpc/CMakeLists.txt]: Local path detected. Please dep. invert. Feel free to use a Find*.cmake script to list possible default paths such as this one but don't include it in the main CMakeLists.txt

File: examples/hpc-c/tvm-hpc/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Notes
+This build is only tested on python3.11.
+
+# Installing dependencies
+## Installing TVM
+Get sources
+``` sh
+git clone --recursive https://github.com/apache/tvm tvm
```
Comment (hegza) [inline examples/hpc-c/tvm-hpc/README.md]: Consider specifying a specific (release?) version. Shallow clone is usually pretty good for large project such as this one if no work on that tree is required (i.e., `--depth=1`)

File: examples/hpc-c/tvm-hpc/tiny_perf_benchmark.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    parser.add_argument("-b", "--benchmark", required=True)
+    opts = parser.parse_args()
+    if opts.benchmark == "kws":
+        run_kws()
+    elif opts.benchmark == "vww":
+        run_vww()
+    elif opts.benchmark == "ic":
+        run_ic()
+    else:
+        print("Bad benchmark! Availeable benchmarks are: kws, vww, ic")
```
Comment (hegza) [inline examples/hpc-c/tvm-hpc/tiny_perf_benchmark.py]: *available

**Review by hegza — APPROVED** [review]
Looks great. It seems the benchmark is based on the serial port, so it works the same for ASIC and VP. Most likely the serial will be the bottleneck for this type of testing so we may need to look into diversifying when the time comes.

File: examples/hpc-c/tvm-hpc/requirements.txt
Comment (vilukissa68) [reply examples/hpc-c/tvm-hpc/requirements.txt:1]: I did `pip freeze` to get exact versions for the packages

File: examples/hpc-c/tvm-hpc/tiny_perf_benchmark.py (hunk 4 above)
Comment (vilukissa68) [reply examples/hpc-c/tvm-hpc/tiny_perf_benchmark.py]: Fixed

File: examples/hpc-c/tvm-hpc/README.md (hunk 3 above)
Comment (vilukissa68) [reply examples/hpc-c/tvm-hpc/README.md]: I didn't think about this, good idea!

File: examples/hpc-c/tvm-hpc/CMakeLists.txt (hunk 2 above)
Comment (vilukissa68) [reply examples/hpc-c/tvm-hpc/CMakeLists.txt]: This include turned out to be unused. It was probably a leftover from before Newlib was fully integrated. Removed.

File: examples/hpc-c/tvm-hpc/README.md (hunk 1 above)
Comment (vilukissa68) [reply examples/hpc-c/tvm-hpc/README.md:58]: 😃

File: examples/hpc-c/tvm-hpc/README.md (hunk 3 above)
Comment (vilukissa68) [reply examples/hpc-c/tvm-hpc/README.md]: Actually the URL is wrong. I have forked TVM in [https://github.com/vilukissa68/tvm-headsail/tree/dnnl-style](https://github.com/vilukissa68/tvm-headsail/tree/dnnl-style). It should probably be hosted on Soc-Hub-Fi.

**Comment by vilukissa68** [comment]
After changes made to the preceding pull requests the acceleration seems to be working with the previously observed accuracy (~50%)

**Pull request merged**